src/commands/package-and-push.yaml

```diff
@@ -35,29 +35,49 @@ steps:
       helm init --client-only
   - run: |
       mkdir build && helm package ./helm/<< parameters.chart >> --destination ./build
-  - run: |
-      helm repo index --url https://giantswarm.github.com/$(cat .app_catalog_name) --merge .app_catalog/index.yaml ./build
-  - run: |
-      cp ./build/* .app_catalog
-  - run: |
-      cd .app_catalog && git add -A
-  - run: |
-      cd .app_catalog && git commit -m "add $(git status --porcelain | cut -c4- | grep << parameters.chart >>)"
   - run:
       name: Push to app catalog repo
       command: |
-          cd .app_catalog && \
-          declare -i attempt=0 && \
-          while ! git push; do
-            attempt+=1
-            if ((attempt >= 4)); then
-              printf '%s\n' \
-                "Error pushing app to the catalog. See known errors in:" \
-                "https://github.com/giantswarm/architect-orb/blob/master/README.md#push-to-app-catalog" \
-                "Giving up after $attempt failures." \
-                >&2
-              exit 1
-            fi
+          readonly app_catalog_name="$(cat .app_catalog_name)"
+
+          cd .app_catalog
+
+          readonly head="$(git rev-parse HEAD)"
+
+          ret=1
+          tries=4
+          for i in $(seq 1 $tries) ; do
+            echo "====> Attempt $i: Running: git reset --hard \$head"
+            git reset --hard $head
+
+            echo "====> Attempt $i: Running: git pull --rebase"
```

Comment:
Do we get anything if we moved `git pull --rebase` right after `git commit` and right before `git push`?

Reply:
No, we don't but I think this flow is more natural even though it costs us one extra `git pull` call. I believe it's extremely efficient though because people do it a lot.

Reply:
If you think different I'm ok with changing that in follow up. I'm merging this now to have it in the release.